sdk/file_sdk/api_impl/stream_handler.cpp

```diff
@@ -9,27 +17,40 @@ DLL_PUBLIC_FILE std::shared_ptr<IStreamHandler> IStreamHandler::Create(std::shar
 }
 
 StreamHandler::StreamHandler(std::shared_ptr<IPolicyEngine> engine, std::shared_ptr<IStream> inputStream, const std::string &inputExtension) {
+  mEngine = engine;
+  mFileFormat = FileFormatFactory::Create(inputStream, inputExtension);
+  mLabels = mFileFormat->GetTags();
 }
 
 bool StreamHandler::IsLabeled() {
-  return false;
+  return mFileFormat->GetTags().size() > 0;
 }
 
 bool StreamHandler::IsProtected() {
   return false;
 }
 
 std::shared_ptr<Tag> StreamHandler::GetLabel() {
-  return nullptr;
+  return std::make_shared<Tag>(mFileFormat->GetTags()[0]);
 }
 
 void StreamHandler::SetLabel(const std::string& labelId, const LabelingOptions& labelingOptions) {
+  std::time_t now = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+  Tag tag(labelId, "Label", "parentId", labelingOptions.GetOwner(), true, std::ctime(&now)); //will get all the missing data such as label name from UPE when impl
+  mLabels.push_back(tag);
```

Comment:
What will happen if you call SetLabel twice?

Reply:
so why is it need to be a vector at all, if it can only hold 1 label at a time?

---
In reply to: [139311427](https://github.com/AzureAD/rms-sdk-for-cpp/pull/153#discussion_r139311427) [](ancestors = 139311427)

Reply:
why are you asking me?